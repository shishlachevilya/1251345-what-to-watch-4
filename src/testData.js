export const movies = [
  {
    id: `1`,
    bg: `bg`,
    title: `Mindhunter`,
    videoPrev: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    preview: `img/mindhunter.jpg`,
    poster: `img/mindhunter.jpg`,
    background: `img/mindhunter.jpg`,
    genre: `Thrillers`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent varius quis lorem id iaculis. Sed eget luctus orci. Donec cursus lectus auctor quam tempus ultricies. Nullam at aliquam massa. Donec mollis quis urna eget condimentum. Mauris at dictum ante. Phasellus maximus massa augue, ut luctus dolor rutrum et. Aliquam faucibus turpis lectus, sed hendrerit dolor malesuada at.`,
    rating: 10.0,
    votes: 32,
    director: `Best Director`,
    starring: [`Nicola Cage`, `Cara Delvin`, `Joseph Moser`],
    duration: 48,
    release: 2016,
    reviews: [
      {
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent varius quis lorem id iaculis. Sed eget luctus orci.`,
        author: `Camille Guaty`,
        rating: 8.4,
        date: `2019-05-08`,
      },
      {
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent varius quis lorem id iaculis. Sed eget luctus orci.`,
        author: `John Dou`,
        rating: 3.4,
        date: `2019-05-08`,
      },
      {
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent varius quis lorem id iaculis. Sed eget luctus orci.`,
        author: `Chris Hemsworth`,
        rating: 5.2,
        date: `2019-05-08`,
      },
      {
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent varius quis lorem id iaculis. Sed eget luctus orci.`,
        author: `Camille Guaty`,
        rating: 8.4,
        date: `2019-05-08`,
      },
      {
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent varius quis lorem id iaculis. Sed eget luctus orci.`,
        author: `John Dou`,
        rating: 3.4,
        date: `2019-05-08`,
      },
      {
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent varius quis lorem id iaculis. Sed eget luctus orci.`,
        author: `Chris Hemsworth`,
        rating: 5.2,
        date: `2019-05-08`,
      }
    ]
  }
];

export const TabLabel = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`
};

export const genreList = [
  `All genres`,
  `Comedies`,
  `Crime`,
  `Documentary`,
  `Dramas`,
  `Horror`,
  `Kids & Family`,
  `Romance`,
  `Sci-Fi`,
  `Thrillers`
];

export default {
  movies,
  TabLabel,
  genreList
};
