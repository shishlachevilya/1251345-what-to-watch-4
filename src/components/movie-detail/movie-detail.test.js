import React from "react";
import renderer from "react-test-renderer";
import MovieDetail from "./movie-detail";
import {genreList, movies} from "../../testData";
import {BrowserRouter as Router} from "react-router-dom";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";

const movie = movies[0];

const mockStore = configureStore([]);

describe(`MovieDetail component test`, () => {
  it(`component should renders correctly`, () => {
    const store = mockStore({
      movies,
      genreList,
      currentGenreItem: genreList[0]
    });

    const tree = renderer
    .create(
        <Provider store={store}>
          <Router>
            <MovieDetail
              activeMovie={movie}
              onCardTitleClick={() => {}}
            />
          </Router>
        </Provider>
    )
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
