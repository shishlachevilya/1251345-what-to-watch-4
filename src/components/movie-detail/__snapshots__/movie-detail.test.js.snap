// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetail component test component should renders correctly 1`] = `
<div>
  <section
    className="movie-card movie-card--full"
    id="1"
    style={
      Object {
        "backgroundColor": "bg",
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Mindhunter"
          src="img/mindhunter.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="main.html"
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <div
            className="user-block__avatar"
          >
            <img
              alt="User avatar"
              height="63"
              src="img/avatar.jpg"
              width="63"
            />
          </div>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Mindhunter
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Thrillers
            </span>
            <span
              className="movie-card__year"
            >
              2016
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <a
              className="btn movie-card__button"
              href="add-review.html"
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Mindhunter"
            height="327"
            src="img/mindhunter.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item movie-nav__item--active"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-rating"
          >
            <div
              className="movie-rating__score"
            >
              10
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                Awesome
              </span>
              <span
                className="movie-rating__count"
              >
                32
                 ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text"
          >
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent varius quis lorem id iaculis. Sed eget luctus orci. Donec cursus lectus auctor quam tempus ultricies. Nullam at aliquam massa. Donec mollis quis urna eget condimentum. Mauris at dictum ante. Phasellus maximus massa augue, ut luctus dolor rutrum et. Aliquam faucibus turpis lectus, sed hendrerit dolor malesuada at.
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    className="page-content"
  >
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="main.html"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
